packages/kbn-unified-data-table/src/types.ts

```diff
@@ -46,10 +46,7 @@ export type DataGridCellValueElementProps = EuiDataGridCellValueElementProps & {
   isCompressed?: boolean;
 };
 
-export type CustomCellRenderer = Record<
-  string,
-  (props: DataGridCellValueElementProps) => ReactElement
->;
+export type CustomCellRenderer = Record<string, FunctionComponent<DataGridCellValueElementProps>>;
```

Comment:
@elastic/kibana-data-discovery 

Changed this to `FunctionComponent` because previous type does not accept memoized components. `FunctionComponent` is more inclusive.

Please let me know if you would keep this a function instead.